src/Command/DataExportCommand.php

```diff
@@ -0,0 +1,215 @@
+        $this->addArgument(self::SALES_CHANNEL_ARGUMENT, InputArgument::OPTIONAL, 'ID of the sales channel');
+        $this->addArgument(self::SALES_CHANNEL_LANGUAGE_ARGUMENT, InputArgument::OPTIONAL, 'ID of the sales channel language');
+    }
+
+    /**
+     * @SuppressWarnings(PHPMD.UnusedPrivateMethod)
+     * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+     */
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        if ($input->isInteractive()) {
+            $helper = $this->getHelper('question');
+
+            $exportTypeQuestion = $this->getChoiceQuestion(sprintf('Select data export type (default  - %s)', self::PRODUCTS_EXPORT_TYPE), [self::PRODUCTS_EXPORT_TYPE, self::CMS_EXPORT_TYPE, self::BRANDS_EXPORT_TYPE], 'Invalid option %s', 0);
+            $exportType         = $helper->ask($input, $output, $exportTypeQuestion);
+
+            $salesChannel     = $this->getSalesChannel($helper->ask($input, $output, new Question('ID of the sales channel (leave empty if no value): ')));
+            $language         = $this->getLanguage($helper->ask($input, $output, new Question('ID of the sales channel language (leave empty if no value): ')));
+
+            $saveFileQuestion = $this->getChoiceQuestion('Save export to local file? (default  - no): ', ['no', 'yes'], 'Invalid option %s', 0);
+            $saveFile         = (bool) array_flip($saveFileQuestion->getChoices())[$helper->ask($input, $output, $saveFileQuestion)];
+
+            $uploadFeedQuestion = $this->getChoiceQuestion('Should upload after exporting? (default  - no): ', ['no', 'yes'], 'Invalid option %s', 0);
+            $uploadFeed         = (bool) array_flip($uploadFeedQuestion->getChoices())[$helper->ask($input, $output, $uploadFeedQuestion)];
+
+            $pushImportQuestion = $this->getChoiceQuestion('Should import after uploading? (default  - no): ', ['no', 'yes'], 'Invalid option %s', 0);
+            $pushImport         = (bool) array_flip($pushImportQuestion->getChoices())[$helper->ask($input, $output, $pushImportQuestion)];
+        } else {
+            $salesChannel     = $this->getSalesChannel($input->getArgument(self::SALES_CHANNEL_ARGUMENT));
+            $language         = $this->getLanguage($input->getArgument(self::SALES_CHANNEL_LANGUAGE_ARGUMENT));
+            $exportType       = $input->getArgument(self::EXPORT_TYPE_ARGUMENT) ?? self::PRODUCTS_EXPORT_TYPE;
+            $uploadFeed       = $input->getOption(self::UPLOAD_FEED_OPTION);
+            $pushImport       = $input->getOption(self::PUSH_IMPORT_OPTION);
+        }
+
+        $context          = $this->channelService->getSalesChannelContext($salesChannel, $language->getId());
+        $entityFQN        = $this->getEntityFqnByType($exportType);
+        $feedService      = $this->feedFactory->create($context, $entityFQN);
+        $feedColumns      = $this->getFeedColumns($exportType, $entityFQN);
+
+        if (isset($saveFile) && $saveFile) {
```

Comment:
I'd set default value to $saveFile = false as then we will not need to check if variable is set